<launch>

  <group ns ="/arm">

    <arg name="prefix" default="/arm"/>
    <!-- <arg name="prefix" default=""/> -->
    <param name="prefix" value="$(arg prefix)"/>

    <arg name="name" default=""/>
    <param name="name" value="$(arg name)"/>

    <arg name="robot_identifier" default="0" />
    <param name="robot_id" value="$(arg robot_identifier)"/>

    <arg name="ip_address" default="192.168.1.3" />
    <param name="ip_address" value="$(arg ip_address)" type="str"/>

    <arg name="use_sim" default="true" />
    <param name="use_sim" value="$(arg use_sim)" type="bool"/>

    <param name="cyclic_time_usec" value="1000"/>
    <!-- <param name="thread_sampling_freq_hz" value="300"/> -->


    <rosparam file="$(find gen3_control)/config/joint_limits.yaml" command="load"/>

    <rosparam file="$(find gen3_control)/config/controller.yaml" command="load" />

    <!-- <param name="robot_description" textfile="$(find gen3_control)/urdf/gen3.urdf"/> -->
    <param name="robot_description" textfile="$(find gen3_control)/urdf/gen3_undress.urdf"/>

    <node name="gen3_control" pkg="gen3_control" type="gen3_control_node" output="screen" />
    <node name="gen3_gui" pkg="gen3_control" type="gen3_gui_node" output="screen" />
    <node name="gen3_planner" pkg="gen3_control" type="gen3_planner_node" output="screen" />
    <node name="gen3_example" pkg="gen3_control" type="gen3_example_node" output="screen" />

    <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false" output="screen" args="
      joint_state_controller
      position_controller
    "/>

  </group>

  <!-- <node name="asar_gui_node" pkg="asar_control" type="asar_gui_node" output="screen" /> -->
  <!-- <remap from="/robotarm_to_grasp" to="/arm/robotarm_to_grasp" /> -->


  <!-- <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false" output="screen" args="
      joint_state_controller
      position_controller
    "/> -->

  <!-- <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" respawn="false" output="screen"> -->
  <!-- <remap from="/joint_states" to="/sim/joint/state" /> -->
  <!-- <remap from="/tf" to="/sim/tf" /> -->
  <!-- </node> -->

</launch>
